packages/pando-lib/src/scripts.ts

```diff
@@ -0,0 +1,248 @@
+import CID from 'cids'
+import IPFS from 'ipfs-http-client'
+import IPLD from 'ipld'
+import IPLDGit from 'ipld-git'
+import { cidToSha } from 'ipld-git/src/util/util.js'
+import orderBy from 'lodash.orderby'
+import uniqWith from 'lodash.uniqwith'
+import multicodec from 'multicodec'
+
+const ipfs = IPFS({ host: 'ipfs.infura.io', port: '5001', protocol: 'https' })
+const ipld = new IPLD({ blockService: ipfs.block, formats: [IPLDGit] })
+
+export const isEqualMultihash = (x: CID, y: CID): boolean => {
+  return (x.toBaseEncodedString() === y.toBaseEncodedString()) && (x.version === y.version)
+}
+
+const fetchHistory = async (cid: CID, history: Commit[]): Promise<Commit[]> => {
+  return new Promise(async (resolve, reject) => {
+    try {
+      const commit = await Commit.get(cid)
+      history.push(commit)
+
+      for (const parent of commit.parents) {
+        history = await fetchHistory(parent, history)
+      }
+
+      resolve(history)
+    } catch (err) {
+      reject(err)
+    }
+  })
+}
+
+export const fetchOrderedHistory = async (cid: CID, formerHistory: Commit[]) => {
+  let history = await fetchHistory(cid, formerHistory)
+  history = uniqWith(history, (one, two) => {
+    return isEqualMultihash(one['@cid'], two['@cid'])
+  })
+
+  history = orderBy(
+    history,
+    [
+      commit => {
+        // https://github.com/git/git/blob/v2.3.0/Documentation/date-formats.txt
+        /* eslint-disable-next-line no-unused-vars */
+        const [timestamp, offset] = commit.author.date.split(' ')
+        return timestamp
+      },
+    ],
+    ['desc']
+  )
+
+  return history
+}
+
+async function fetchFilesFromTree(path: string, cid: CID, treeObj: any): Promise<IModifiedTree> {
+  return new Promise<IModifiedTree>( async (resolve, reject) => {
+    try {
+      const modifiedTree = treeObj ? treeObj : {}
+      let result = ipld.tree(cid, path, { recursive: true })
+      path = await result.first() // What to do if we get an array of paths...
+    
+      result = ipld.resolve(cid, path)
+      const paths = await result.all()
+    
+      paths.map(file => {
+        if (file.remainderPath === '') {
+          modifiedTree[path] = { mode: file.value.mode, blob: file.value.hash }
+        }
+        else {
+          fetchFilesFromTree(file.remainderPath, file.value, modifiedTree)
+        }
+      })
+      resolve(modifiedTree)
+    }
+    catch(err) {
+      reject(err)
+    }
+  })
+}
+
+export interface IAuthorOrCommitter {
+  name: string
+  email: string
+  date: string
+}
+
+export interface ITree { [path: string]: string }
```

Comment:
What is this for ? It's not used anywhere.

Reply:
Yes I will remove